app/src/main/java/de/westnordost/streetcomplete/quests/step_count/AddStepCountStile.kt

```diff
@@ -0,0 +1,52 @@
+package de.westnordost.streetcomplete.quests.step_count
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmElementQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.OUTDOORS
+
+class AddStepCountStile : OsmElementQuestType<Int> {
+
+    private val stileNodeFilter by lazy { """
+        nodes with barrier=stile
+         and stile ~ stepover|ladder
+         and access !~ private|no
+         and !step_count
+    """.toElementFilterExpression() }
+
+    private val excludedWaysFilter by lazy { """
+        ways with
+          access ~ private|no
+          and foot !~ permissive|yes|designated
+    """.toElementFilterExpression() }
+
+    override fun getApplicableElements(mapData: MapDataWithGeometry): Iterable<Element> {
+        val excludedWayNodeIds = mutableSetOf<Long>()
+        mapData.ways
+            .filter { excludedWaysFilter.matches(it) }
+            .flatMapTo(excludedWayNodeIds) { it.nodeIds }
```

Comment:
Does this work as well?


```suggestion
        val excludedWayNodeIds = mapData.ways
            .filter { excludedWaysFilter.matches(it) }
            .flatMap { it.nodeIds }
```

Reply:
Yep, tested by having solely `        val excludedWayNodeIds = mutableSetOf<Long>()` and failing tests and then switching to this form.

This code existed in several places. Hmmm, maybe it should be extracted to some function?

Reply:
Uh, what is this?

Isn't `excludedWayNodeIds` now not just a list?

This'd wrong for two reasons:
1. it then can contain duplicate node ids. Yes, this is completely possible, e.g. two ways that cross share a node
2. more importantly, it is deliberate that this collection is a **Set** because `contains()` is executes in O(1) while on a List it executes in O(n)